util/settings/settings.go

```diff
@@ -812,13 +816,14 @@ func (mgr *SettingsManager) SaveTLSCertificateData(ctx context.Context, tlsCerti
 }
 
 // NewSettingsManager generates a new SettingsManager pointer and returns it
-func NewSettingsManager(ctx context.Context, clientset kubernetes.Interface, namespace string) *SettingsManager {
+func NewSettingsManager(ctx context.Context, clientset kubernetes.Interface, namespace string, disableAdmin bool) *SettingsManager {
```

Comment:
I suggest passing `disableAdmin` as a parameter of `InitializeSettings` method instead of adding it to the `SettingManager` data structure. So we don't have to pass default `false` in every place where settings manager is used.